### PR TITLE
Get the prelude working

### PR DESCRIPTION
**Pull request by ayberkt**

### CODE REVIEW

File: src/plutus-core-execution.k [hunk 1]
```diff
@@ -1,99 +1,87 @@
...
+    => Float2Int(roundFloat(#toFloat(F), 2, 2))
+
+  syntax PlFloat ::= #toPlFloat(Float)                                [function]
+  rule #toPlFloat(F:Float)
+    => #parseToken("PlFloat@PLUTUS-CORE-SYNTAX", Float2String(F))
+  rule #toFloat(F:PlFloat)
+    => #parseToken("Float@FLOAT-SYNTAX", #toStringFloat(F))
 
   syntax ByStr ::= toByStrVerbatim ( String )                        [function]
   rule toByStr(S:String)
```
Comment (lucaspena) [inline src/plutus-core-execution.k]: `toByStrVerbatim` is declared but never used, while `toByStr` is used but never declared

Also, these should begin with a `#`

File: src/plutus-core-execution.k [hunk 2]
```diff
@@ -102,57 +90,85 @@ module PLUTUS-CORE-EXECUTION
...
+  rule M:Val ~> #appFreeze(MS) => [ M MS ]
+  rule [ [ F:Val M:Tm ] MS:TmList ] => [ F M MS ]
+
+  // When a lambda abstracts more than one variable, desugar it to a
+  // combination of unary lambdas.
+  rule [ ( lam X1:Var X2:Var XS:VarList M:Tm ) N:Tm ]
+    => [ ( lam X1 .VarList ( lam X2 XS M ) ) N ]
+  // In the unary case, just perform the substitution.
+  rule [ ( lam X:Var .VarList M:Tm ) N:Tm ] => M[N / X]
+
```
Comment (lucaspena) [inline src/plutus-core-execution.k]: I still don't love how application is handled. I think we should try to handle it more like the nested lambdas in the sense that we desugar it to explicit unary application.

File: src/plutus-core-execution.k [hunk 3]
```diff
@@ -102,57 +90,85 @@ module PLUTUS-CORE-EXECUTION
...
   rule (.DefList ~> _:ModF ) => .K
 
   /***************************************************************************/
   /************************   PATTERN MATCHING   *****************************/
   /***************************************************************************/
 
-  syntax Map ::= manySubst(VarList, ValList) [function]
+  syntax Map ::= manySubst(VarList, ValList)                         [function]
+
```
Comment (lucaspena) [inline src/plutus-core-execution.k]: change to `#manySubst`

File: src/plutus-core-syntax.k [hunk 4]
```diff
@@ -22,7 +22,8 @@ module PLUTUS-CORE-SYNTAX
   // expression as that in the Plutus Core specification.
 
   /* TODO: Make sure that K's built-in Float is compatible with PlFloat*/
```
Comment (lucaspena) [inline src/plutus-core-syntax.k]: Comment should be updated. Maybe `TODO: add hook to ensure all floats are always of the correct format`

File: src/plutus-core-syntax.k [hunk 5]
```diff
@@ -112,11 +116,13 @@ module PLUTUS-CORE-SYNTAX
...
+  syntax KResult    ::= Val
+  syntax KVariable  ::= Var
 
-  syntax String       ::= toStringVerbatim(ByStr)  [ function
+  syntax String     ::= toStringVerbatim(ByStr)    [ function
+                                                   , hook(STRING.token2string)]
+                      | #toStringFloat(PlFloat)     [ function
                                                    , hook(STRING.token2string)]
-                        | toString(ByStr)                            [function]
+                      | toString(ByStr)                              [function]
```
Comment (lucaspena) [inline src/plutus-core-syntax.k]: Change to `#toStringVerbatim` and `#toString`

**Comment by ayberkt** [comment]
💯

**Pull request merged**